### ISSUE TITLE
Gitops process fails for repositories that have approval requirements

### ISSUE THREAD

**Issue by seansund** [cloud-native-toolkit/software-everywhere#351]
In production environments, pull requests and approvals are often required before changes can be merged into the main branch. Currently the gitops automation creates pull requests (in most cases) but tries to auto-merge the pull request immediately after it is created. If the environment has restrictions in place, the pull request will be in `blocked` status until this is resolved.

1. All places where changes are made to the gitops repository should be done in a pull request
    - [ ] git-repo module - repo initialization happens with direct push
    - [ ] gitops-repo module - gitops structure initialization happens with direct push
2. git-client should include delay if pull request is in blocked state to allow the PR to be manually unblocked (approvals, etc)
